src/serial.rs

```diff
@@ -377,11 +377,14 @@ impl<T: Trigger, EV: SerialEvents, W: Write> Serial<T, EV, W> {
                         self.received_data_interrupt().map_err(Error::Trigger)?;
                     }
                 } else {
-                    self.out.write_all(&[value]).map_err(Error::IOError)?;
-                    self.out.flush().map_err(Error::IOError)?;
-                    self.serial_evts.out_byte();
-
+                    let res = self
+                        .out
+                        .write_all(&[value])
+                        .map_err(Error::IOError)
```

Comment:
What do you think about adding a new event to `SerialEvents`, `tx_lost_byte()` (another better name is welcome), which will be called in case of `write` error?